agent/command/papertrail_trace.go

```diff
@@ -0,0 +1,146 @@
+package command
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/evergreen-ci/evergreen/agent/internal"
+	"github.com/evergreen-ci/evergreen/agent/internal/client"
+	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+type papertrailTrace struct {
+	Address   string   `mapstructure:"address" plugin:"expand"`
+	KeyID     string   `mapstructure:"key_id" plugin:"expand"`
+	SecretKey string   `mapstructure:"secret_key" plugin:"expand"`
+	Product   string   `mapstructure:"product" plugin:"expand"`
+	Version   string   `mapstructure:"version" plugin:"expand"`
+	Filenames []string `mapstructure:"filenames" plugin:"expand"`
+	WorkDir   string   `mapstructure:"work_dir" plugin:"expand"`
+	base
+}
+
+// This command is owned by the Dev Prod Release Infrastructure team
+func (t *papertrailTrace) Execute(ctx context.Context,
+	comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
+	if err := util.ExpandValues(t, &conf.Expansions); err != nil {
+		return errors.Wrap(err, "apply expansions")
```

Comment:
errors in this function need to be re-written too

Reply:
Done